extension/src/openvic-extension/singletons/MapItemSingleton.cpp

```diff
@@ -0,0 +1,251 @@
+		if (obj->is_type<GFX::Billboard>()) {
+			add_billboard_dict(obj->get_name(), ret);
+		}
+	}
+	
+	return ret;
+}
+
+//dont use this, it includes water provinces
+int32_t MapItemSingleton::get_province_count() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, 0);
+
+	return game_singleton->get_definition_manager().get_map_definition().get_province_definition_count();
+}
+
+PackedVector2Array MapItemSingleton::get_province_positions() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedVector2Array());
+
+	PackedVector2Array billboard_pos {};
+	
+	for(ProvinceDefinition const& prov : game_singleton->get_definition_manager().get_map_definition().get_province_definitions()){
+		if(prov.is_water()) continue; //billboards dont appear over water, skip
+
+		fvec2_t city_pos = prov.get_city_position();
+		Vector2 pos = Utilities::to_godot_fvec2(city_pos) / game_singleton->get_map_dims();
+		billboard_pos.push_back(pos);
+
+	}
+	
+	return billboard_pos;
+}
+
+//includes non-existent countries, used for setting the billboard buffer size
+int32_t MapItemSingleton::get_capital_count() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, 0);
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, 0);
+
+	return instance_manager->get_country_instance_manager().get_country_instance_count();
+}
+
+PackedVector2Array MapItemSingleton::get_capital_positions() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedVector2Array());
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, PackedVector2Array());
+
+	PackedVector2Array billboard_pos {};
+
+	for(CountryInstance const& country : instance_manager->get_country_instance_manager().get_country_instances()){
+		if(!country.exists()) continue; //skip non-existant countries
+
+		fvec2_t city_pos = country.get_capital()->get_province_definition().get_city_position();
+		Vector2 pos = Utilities::to_godot_fvec2(city_pos) / game_singleton->get_map_dims();
+		billboard_pos.push_back(pos);
+
+	}
+
+	return billboard_pos;
+}
+
+PackedByteArray MapItemSingleton::get_crime_icons() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, PackedByteArray());
+
+	InstanceManager const* instance_manager = game_singleton->get_instance_manager();
+	ERR_FAIL_NULL_V(instance_manager, PackedByteArray());
+
+	PackedByteArray icons {};
+
+	for(ProvinceInstance const& prov_inst : instance_manager->get_map_instance().get_province_instances()){
+		ProvinceDefinition const* prov = &prov_inst.get_province_definition();
+		if(prov->is_water()) continue; //billboards dont appear over water, skip
```

Comment:
```suggestion
		if (prov_inst.get_province_definition().is_water()) continue; //billboards dont appear over water, skip
```
- `get_province_definition()` returns a `const&`, so no need to turn it into a `const*`
- Since this is only used to check if it's a water province, might as well use it directly rather than creating a variable (this actually makes it take fewer CPU cycles too)
- Same for `get_rgo_icons` and `get_national_focus_icons` below